node/src/blockchain/blockchain_bridge.rs

```diff
@@ -426,223 +358,242 @@ mod tests {
-            )
-        );
-
-        let result = request.wait().unwrap().unwrap();
-        let mut expected_payment_0 = Payment::new(
-            make_wallet("blah"),
-            42,
-            H256::from("sometransactionhash".keccak256()),
-        );
-
-        if let Ok(zero) = result.clone().get(0).unwrap().clone() {
-            assert!(
-                zero.timestamp
-                    <= expected_payment_0
-                        .timestamp
-                        .checked_add(Duration::from_secs(2))
-                        .unwrap()
-            );
-            assert!(
-                zero.timestamp
-                    >= expected_payment_0
-                        .timestamp
-                        .checked_sub(Duration::from_secs(2))
-                        .unwrap()
-            );
-            expected_payment_0.timestamp = zero.timestamp
-        }
-
-        let mut expected_payment_1 = Payment::new(
-            make_wallet("foo"),
-            21,
-            H256::from("someothertransactionhash".keccak256()),
+            *send_transaction_params,
+            vec![
+                (
+                    consuming_wallet.clone(),
+                    make_wallet("blah"),
+                    420,
+                    U256::from(1),
+                    expected_gas_price
+                ),
+                (
+                    consuming_wallet.clone(),
+                    make_wallet("foo"),
+                    210,
+                    U256::from(2),
+                    expected_gas_price
+                )
+            ]
         );
-
-        if let Ok(one) = result.clone().get(1).unwrap().clone() {
-            assert!(
-                one.timestamp
-                    <= expected_payment_1
-                        .timestamp
-                        .checked_add(Duration::from_secs(2))
-                        .unwrap()
-            );
-            assert!(
-                one.timestamp
-                    >= expected_payment_1
-                        .timestamp
-                        .checked_sub(Duration::from_secs(2))
-                        .unwrap()
-            );
-            expected_payment_1.timestamp = one.timestamp
-        }
-
-        assert_eq!(result[1], Ok(expected_payment_1));
-
+        let get_transaction_count_params = get_transaction_count_params_arc.lock().unwrap();
         assert_eq!(
-            transaction_count_parameters.lock().unwrap()[0],
-            consuming_wallet.clone(),
+            *get_transaction_count_params,
+            vec![consuming_wallet.clone(), consuming_wallet.clone()]
         );
+        let accountant_received_payment = accountant_recording_arc.lock().unwrap();
+        assert_eq!(accountant_received_payment.len(), 1);
+        let sent_payment_msg = accountant_received_payment.get_record::<SentPayments>(0);
+        let mut sent_payments_cloned = sent_payment_msg.payments.clone();
+        sent_payments_cloned
```

Comment:
Pull the actual timestamps out into a separate list here, before replacing them with 0, and then assert separately on the list of timestamps that all of them are between when the Act started and when it finished.

Also, replace the `if let` with an `unwrap()` so that if there is an error in the list, we get an immediate failure. (Unless there are meant to be errors in the list, of course: then assert that they're what they should be.)